Source/WebCore/rendering/RenderLayerBacking.cpp

```diff
@@ -2596,9 +2611,28 @@ bool RenderLayerBacking::updateTransformFlatteningLayer(const RenderLayer* compo
     return layerChanged;
 }
 
-bool RenderLayerBacking::updateViewportConstrainedAnchorLayer(bool needsAnchorLayer)
+bool RenderLayerBacking::updateViewportConstrainedSublayers(ViewportConstrainedSublayers sublayers)
 {
     bool layerChanged = false;
+
+    using enum ViewportConstrainedSublayers;
+
+    bool needsAnchorLayer = false;
+    bool needsClippingLayer = false;
+    switch (sublayers) {
+    case NoLayers:
+        break;
+    case AnchorLayer: {
+        needsAnchorLayer = true;
+        break;
+    }
+    case ClippingAndAnchorLayers: {
+        needsAnchorLayer = true;
+        needsClippingLayer = true;
+        break;
+    }
+    }
```

Comment:
Maybe some fallthrough so that DRY? Feel free to ignore if you think it's confusing.

Also, you don't need new scopes if you aren't declaring variables.

```suggestion
    switch (sublayers) {
    case NoLayers:
        break;
    case ClippingAndAnchorLayers:
        needClippingLayer = true;
    case AnchorLayer:
        needsAnchorLayer = true;
        break;
    }
```

Reply:
Sounds good to me — will update this switch case.

Thanks for the review!